sssd_test_framework/markers.py

```diff
@@ -22,10 +24,17 @@ def pytest_configure(config: pytest.Config):
     )
 
 
-def builtwith(item: pytest.Function, requirements: dict[str, str], **kwargs: BaseRole):
+def builtwith(item: pytest.Function, requirements: dict[str, str], topology: KnownTopology, **kwargs: BaseRole):
     def value_error(msg: str) -> ValueError:
         return ValueError(f"{item.nodeid}::{item.originalname}: @pytest.mark.builtwith: {msg}")
 
+    mh = item.funcargs["mh"]
+
+    # Skip feature detection in topologies that are not running
+    # Always check feature detection in client
+    if "client" not in requirements and topology and topology.value.topology != mh.topology:
```

Comment:
Do not specialize client. Rather check that `topology is None or not matching`

Reply:
Done